mesonbuild/modules/features/feature.py

```diff
@@ -0,0 +1,356 @@
+# Copyright (c) 2023, NumPy Developers.
+# All rights reserved.
+import re
+from typing import (
+    Dict, Set, Tuple, List, Callable, Optional,
+    Union, Any, Iterable, cast, TYPE_CHECKING
+)
```

Comment:
We don't do this anywhere else in Meson, we do `import typing as T` and then use the `T` namespace. For consistency I'd prefer if we kept with that.